File: gcc/testsuite/gfortran.dg/gomp/appendix-a/a.32.1.f90
! { dg-do compile }
! { dg-require-effective-target tls }

      MODULE M
        REAL, POINTER, SAVE :: WORK(:)
        INTEGER :: SIZE
        REAL :: TOL
!$OMP THREADPRIVATE(WORK,SIZE,TOL)
      END MODULE M
      SUBROUTINE A32( T, N )
        USE M
        REAL :: T
        INTEGER :: N
        TOL = T
        SIZE = N
!$OMP PARALLEL COPYIN(TOL,SIZE)
        CALL BUILD
!$OMP END PARALLEL
      END SUBROUTINE A32
      SUBROUTINE BUILD
        USE M
        ALLOCATE(WORK(SIZE))
        WORK = TOL
      END SUBROUTINE BUILD
! { dg-final { cleanup-modules "M" } }

File: gcc/testsuite/gcc.dg/gomp/appendix-a/a.32.1.c
/* { dg-do compile } */
/* { dg-require-effective-target tls } */

#include <stdlib.h>
float *work;
int size;
float tol;
void build (void);
#pragma omp threadprivate(work,size,tol)
void
a32 (float t, int n)
{
  tol = t;
  size = n;
#pragma omp parallel copyin(tol,size)
  {
    build ();
  }
}
void
build ()
{
  int i;
  work = (float *) malloc (sizeof (float) * size);
  for (i = 0; i < size; ++i)
    work[i] = tol;
}
